### PR TITLE
Hono JMeter Plugin: using mean value of all messages delivery time in Hono Receiver

### PR DESCRIPTION
**Pull request by balsmn**
This is the second PR following #357 (as agreed in #338)

Whenever a message with timestamp is received (timestamp either in payload or as application property), message delivery is calculated (as _receivedTime - timeInPayload_) and added to total delivery time.

Now when sample is called, the sampling time for Hono Receiver is calculated as follows:

_totalSampleDeliveryTime  / messageReceivedCount_

Signed-off-by: Balasubramanian <balasubramanian.azhagappan@bosch-si.com>

### CODE REVIEW

**Comment by balsmn** [comment]
@pellmann Could you review this?

File: jmeter/src/main/java/org/eclipse/hono/jmeter/client/HonoReceiver.java [hunk 1]
```diff
@@ -154,10 +156,14 @@ private void noMessagesReceived(final SampleResult result) {
         result.setIdleTime(0);
     }
 
-    private void verifySenderTimeAndSetSamplingTime(final Long time, boolean senderTimeInPayload) {
-        if (time != null) {
-            sampleStart = time;
-            LOGGER.debug("Message sent time : {}", sampleStart);
+    private void verifySenderTimeAndSetSamplingTime(final Long senderTime, long sampleReceivedTime) {
+        if (senderTime != null) {
+            if (sampleStart != 0) { // set sample start once the first message is received.
```
Comment (pellmann) [inline jmeter/src/main/java/org/eclipse/hono/jmeter/client/HonoReceiver.java]: I think this should be sampleStart==0?

File: jmeter/src/main/java/org/eclipse/hono/jmeter/client/HonoReceiver.java [hunk 2]
```diff
@@ -51,6 +51,7 @@
     private long sampleStart;
     private long sampleEnd;
```
Comment (pellmann) [inline jmeter/src/main/java/org/eclipse/hono/jmeter/client/HonoReceiver.java]: sampleEnd is no longer used

Comment (balsmn) [reply jmeter/src/main/java/org/eclipse/hono/jmeter/client/HonoReceiver.java]: Yes. I will remove it

File: jmeter/src/main/java/org/eclipse/hono/jmeter/client/HonoReceiver.java (hunk 1 above)
Comment (balsmn) [reply jmeter/src/main/java/org/eclipse/hono/jmeter/client/HonoReceiver.java]: yes

**Comment by sophokles73** [comment]
@pellmann any news on this one? Seems like Bala made some changes ...

**Pull request merged**